llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -242,8 +242,16 @@ class PGOCounterPromoter {
     if (!isPromotionPossible(&L, LoopExitBlocks))
       return;
 
+    auto IsSuspendBB = [&](BasicBlock *BB) {
```

Comment:
can this be a common utility function somewhere such as BasicBlockUtils.h?

Reply:
This is a perfect segway to whether we should handle the suspend/switch pattern somewhere more centrally - I wouldn't be surprised other analyses would make incorrect inferences about the 'suspend (i.e. ret)' case.

Reply:
In fact, there are other passes failed to handle the suspend intrinsic. Originally we were wondering if we can fix the issues by providing a new analysis pass (e.g., CoroInfoPass). But we failed to go on (due to limited development resources). So it sounds a good idea to extract the common patterns to common utilities. Although I don't think this should be a requirement for this specific PR.